client/verta/verta/_registry/modelversion.py

```diff
@@ -188,61 +188,12 @@ def _create_proto_internal(cls, conn, ctx, name, desc=None, tags=None, attrs=Non
-
-        Parameters
-        ----------
-        model : str or file-like or object
-            Model or some representation thereof.
-                - If str, then it will be interpreted as a filesystem path, its contents read as bytes,
-                  and uploaded as an artifact. If it is a directory path, its contents will be zipped.
-                - If file-like, then the contents will be read as bytes and uploaded as an artifact.
-                - Otherwise, the object will be serialized and uploaded as an artifact.
-        custom_modules : list of str, optional
-            Paths to local Python modules and other files that the deployed model depends on.
-                - If directories are provided, all files within—excluding virtual environments—will
-                  be included.
-                - If not provided, all Python files located within `sys.path`—excluding virtual
-                  environments—will be included.
-        model_api : :class:`~utils.ModelAPI`, optional
-            Model API specifying details about the model and its deployment.
-        artifacts : list of str, optional
-            Keys of logged artifacts to be used by a class model.
-        overwrite : bool, default False
-            Whether to allow overwriting an existing artifact with key `key`.
-
-        """
         if self.has_model and not overwrite:
             raise ValueError("model already exists; consider setting overwrite=True")
 
-        if (artifacts is not None
-                and not (isinstance(artifacts, list)
-                         and all(isinstance(artifact_key, six.string_types) for artifact_key in artifacts))):
-            raise TypeError("`artifacts` must be list of str, not {}".format(type(artifacts)))
-
-        # validate that `artifacts` are actually logged
-        if artifacts:
-            self._refresh_cache()
-            run_msg = self._msg
-            existing_artifact_keys = {artifact.key for artifact in run_msg.artifacts}
-            unlogged_artifact_keys = set(artifacts) - existing_artifact_keys
-            if unlogged_artifact_keys:
-                raise ValueError("`artifacts` contains keys that have not been logged: {}".format(sorted(unlogged_artifact_keys)))
-
-        # associate artifact dependencies
-        if artifacts:
-            self.add_attribute(_MODEL_ARTIFACTS_ATTR_KEY, artifacts, overwrite=overwrite)
-
-        if isinstance(model, six.string_types):  # filepath
-            model = open(model, 'rb')
-        serialized_model, method, model_type = _artifact_utils.serialize_model(model)
-
-        try:
-            extension = _artifact_utils.get_file_ext(serialized_model)
-        except (TypeError, ValueError):
-            extension = _artifact_utils.ext_from_method(method)
+        super(RegisteredModelVersion, self).log_model(model, custom_modules, model_api, artifacts, overwrite)
 
+    def _log_model_as_artifact(self, serialized_model, extension, method, overwrite):
```

Comment:
What would you think about just calling this `_log_model()`? 
`_log_model_as_artifact()` is a bit misleading, since there's nothing else you could log a model as, anyway.

Reply:
Fixed.